### ISSUE TITLE
Roles - Grants; Best practice question

### ISSUE THREAD

**Issue by ben-shory** [permitio/rebac-rego-example#1]
Checking if it considered common or even a best practice to have grants associated with roles like in roles.json.
Isn't that spreading the logic outside of the rego policy?

**Comment by orweis** [comment]
There's always a tradeoff to be made between data for policy as code, and policy as code (it is btw the same tradeoff as data or configuration for application). A tradeoff between velocity of change and ease of auditing / versioning. 

For your use case you should pick the right balance point between speed and ease of auditing. 
You should always have some core defined as code. And you should prefer that elements that require auditing are versioned as code - even if are not stored as code.

In short: yes this is common, and a fair balance point.
